### PR TITLE
Adding CLI subcommands for check, certify and support

### PR DESCRIPTION
**Pull request by tonytcampbell**
Adding three subcommands to the CLI
Check: Runs the checks.  Used for iterative development and fixes
Certify: Will send the results to Red Hat (not implemented)
Support: Will take the failing cert results and logs and create a support ticket (not implemented)

### CODE REVIEW

File: certification/engine/engine.go
Comment (samira-barouti) [inline certification/engine/engine.go:128]: Same comment here

File: certification/engine/engine.go
Comment (samira-barouti) [inline certification/engine/engine.go:91]: We can get rid of nameToCheckMap initialization (line 76). This var should be initialized with either `containerPolicyChecks` or `operatorPolicyChecks` based on the flag, set by the consumer.

File: certification/engine/engine.go
Comment (tonytcampbell) [reply certification/engine/engine.go:117]: I am intentionally introducing a new concept.  Policy == a collection of checks.  This is why I wanted to change what we were originally calling Policies to checks, so that we could use the term Policy to refer to a group of checks. This will allow Product Managers to define a Policy with Checks underneath it. Then when you use our tool you can pass in a subcommand stating which Policy (Group of checks) you want to get certified for.  Some checks may belong to multiple policies.

File: certification/engine/engine.go
Comment (tonytcampbell) [reply certification/engine/engine.go:128]: comment above

File: certification/engine/engine.go
Comment (tonytcampbell) [reply certification/engine/engine.go:91]: Agreed.  Was planning to do that in a future PR.

File: cmd/support.go
Comment (jomkz) [inline cmd/support.go:15]: I know that this is just a placeholder, but should it print "Support command" here?

**Review by jomkz — CHANGES_REQUESTED** [review]

File: cmd/support.go
Comment (tonytcampbell) [reply cmd/support.go:15]: Yes.  I'll change that.

**Comment by komish** [comment]
@tonytcampbell just a ping that this needs conflict resolution

File: Makefile
Comment (bcrochet) [inline Makefile:8]: Can we move the ldflags to a variable to shorten this line? I can add a separate PR if that helps.

File: certification/engine/engine.go
Comment (bcrochet) [inline certification/engine/engine.go:118]: Looks like this could be a small utility function. Keep the exported funcs, but use a utility func and pass in the list of checks.

**Review by bcrochet — COMMENTED** [review]
Couple of small nits. Neither will hold up merge/approval for me. Can be addressed in future PR.

File: certification/engine/engine.go
Comment (tonytcampbell) [reply certification/engine/engine.go:118]: In a future PR, I'm planning to do something like func `GetChecks(policy string) []string.`  Was planning to knock it out once this merges.

File: Makefile
Comment (tonytcampbell) [reply Makefile:8]: Yes.  A separate PR will be great.  Thanks.

**Review by jomkz — APPROVED** [review]

**Comment by jomkz** [comment]
I am good with this but can we squash the commits?

**Comment by tonytcampbell** [comment]
> I am good with this but can we squash the commits?

@jomkz I figured I test the cool Squash on merge feature in GitHub.  (ok, so truth is I forgot to squash but I do want to try out the feature.)

**Comment by tonytcampbell** [comment]
Ok....I squashed it.  No PR approval love besides @jomkz so maybe the Squash will get me a second approval.  Lol.

File: cmd/check_container.go
Comment (samira-barouti) [inline cmd/check_container.go:46]: You can remove the logger from the arg list.

File: cmd/check_operator.go
Comment (samira-barouti) [inline cmd/check_operator.go:46]: same as above

File: cmd/check_container.go
Comment (tonytcampbell) [reply cmd/check_container.go]: Thanks @samira-barouti .  Done

File: cmd/check_operator.go
Comment (tonytcampbell) [reply cmd/check_operator.go]: Done

**Review by samira-barouti — APPROVED** [review]

**Pull request merged**
